ext/opentelemetry-ext-opentracing-shim/tests/test_shim.py

```diff
@@ -431,3 +443,77 @@ def test_span_on_error(self):
         self.assertEqual(
             scope.span.unwrap().events[0].attributes["error.kind"], Exception
         )
+
+    def test_inject(self):
+        """Test `inject()` method."""
+
+        otel_context = trace.SpanContext(trace_id=1220, span_id=7478)
+        context = opentracingshim.SpanContextShim(otel_context)
+
+        # Verify Format.TEXT_MAP
+        text_map = {}
+        self.shim.inject(context, opentracing.Format.TEXT_MAP, text_map)
+        self.assertEqual(text_map[MockHTTPTextFormat.TRACE_ID_KEY], str(1220))
+        self.assertEqual(text_map[MockHTTPTextFormat.SPAN_ID_KEY], str(7478))
+
+        # Verify Format.HTTP_HEADERS
+        http_headers = {}
+        self.shim.inject(
+            context, opentracing.Format.HTTP_HEADERS, http_headers
+        )
+        self.assertEqual(
+            http_headers[MockHTTPTextFormat.TRACE_ID_KEY], str(1220)
+        )
+        self.assertEqual(
+            http_headers[MockHTTPTextFormat.SPAN_ID_KEY], str(7478)
+        )
+
+        # Verify exception for non supported binary format.
+        with self.assertRaises(opentracing.UnsupportedFormatException):
+            self.shim.extract(opentracing.Format.BINARY, bytearray())
+
+    def test_extract(self):
+        """Test `extract()` method."""
+
+        headers = {
+            MockHTTPTextFormat.TRACE_ID_KEY: 1220,
+            MockHTTPTextFormat.SPAN_ID_KEY: 7478,
+        }
+
+        # Verify Format.TEXT_MAP
+        ctx_text_map = self.shim.extract(opentracing.Format.TEXT_MAP, headers)
+        self.assertEqual(ctx_text_map.unwrap().trace_id, 1220)
+        self.assertEqual(ctx_text_map.unwrap().span_id, 7478)
+
+        # Verify Format.HTTP_HEADERS
+        ctx_http = self.shim.extract(opentracing.Format.HTTP_HEADERS, headers)
+        self.assertEqual(ctx_http.unwrap().trace_id, 1220)
+        self.assertEqual(ctx_http.unwrap().span_id, 7478)
+
+        # Verify exception for non supported binary format.
+        with self.assertRaises(opentracing.UnsupportedFormatException):
+            self.shim.extract(opentracing.Format.BINARY, bytearray())
+
+
+class MockHTTPTextFormat(HTTPTextFormat):
```

Comment:
It is not that hard actually but I wanted to keep these tests independently from actual  implementations. Of course I am open to discuss why it could be better to test with the actual implementation instead.

Reply:
Sounds good to me, but lets wait for others to chime in ;)